nixos/modules/services/networking/seafile.nix

```diff
@@ -162,137 +299,250 @@ in
+          RuntimeDirectory = "seafile";
+          LogsDirectory = "seafile";
+          ConfigurationDirectory = "seafile";
+          ReadWritePaths = lib.lists.optional (cfg.dataDir != "${seafRoot}/data") cfg.dataDir;
         };
       in
       {
         seaf-server = {
           description = "Seafile server";
           partOf = [ "seafile.target" ];
-          after = [ "network.target" ];
+          unitConfig.RequiresMountsFor = lib.lists.optional (cfg.dataDir != "${seafRoot}/data") cfg.dataDir;
+          requires = [ "mysql.service" ];
+          after = [
+            "network.target"
+            "mysql.service"
+          ];
           wantedBy = [ "seafile.target" ];
-          restartTriggers = [ ccnetConf seafileConf ];
-          path = [ pkgs.sqlite ];
-          serviceConfig = securityOptions // {
-            User = "seafile";
-            Group = "seafile";
-            DynamicUser = true;
-            StateDirectory = "seafile";
-            RuntimeDirectory = "seafile";
-            LogsDirectory = "seafile";
-            ConfigurationDirectory = "seafile";
+          restartTriggers = [
+            ccnetConf
+            seafileConf
+          ];
+          serviceConfig = serviceOptions // {
             ExecStart = ''
-              ${cfg.seafilePackage}/bin/seaf-server \
+              ${lib.getExe cfg.seahubPackage.seafile-server} \
               --foreground \
               -F /etc/seafile \
               -c ${ccnetDir} \
-              -d ${dataDir} \
+              -d ${cfg.dataDir} \
               -l /var/log/seafile/server.log \
               -P /run/seafile/server.pid \
               -p /run/seafile
             '';
           };
           preStart = ''
             if [ ! -f "${seafRoot}/server-setup" ]; then
-                mkdir -p ${dataDir}/library-template
-                mkdir -p ${ccnetDir}/{GroupMgr,misc,OrgMgr,PeerMgr}
-                sqlite3 ${ccnetDir}/GroupMgr/groupmgr.db ".read ${cfg.seafilePackage}/share/seafile/sql/sqlite/groupmgr.sql"
-                sqlite3 ${ccnetDir}/misc/config.db ".read ${cfg.seafilePackage}/share/seafile/sql/sqlite/config.sql"
-                sqlite3 ${ccnetDir}/OrgMgr/orgmgr.db ".read ${cfg.seafilePackage}/share/seafile/sql/sqlite/org.sql"
-                sqlite3 ${ccnetDir}/PeerMgr/usermgr.db ".read ${cfg.seafilePackage}/share/seafile/sql/sqlite/user.sql"
-                sqlite3 ${dataDir}/seafile.db ".read ${cfg.seafilePackage}/share/seafile/sql/sqlite/seafile.sql"
-                echo "${cfg.seafilePackage.version}-sqlite" > "${seafRoot}"/server-setup
+                mkdir -p ${cfg.dataDir}/library-template
+                # Load schema on first install
+                ${pkgs.mysql-client}/bin/mysql --database=ccnet_db < ${cfg.seahubPackage.seafile-server}/share/seafile/sql/mysql/ccnet.sql
+                ${pkgs.mysql-client}/bin/mysql --database=seafile_db < ${cfg.seahubPackage.seafile-server}/share/seafile/sql/mysql/seafile.sql
```

Comment:
mysql-client is an alias to mariadb.client and we should avoid using aliases. Normally ofborg should fail on those.